test/solver/idr_kernels.cpp

```diff
@@ -40,6 +40,11 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <gtest/gtest.h>
 
 
+#ifdef GKO_COMPILING_DPCPP
+#include <CL/sycl.hpp>
+#endif
```

Comment:
```cpp
if (exec->get_queue()->get_device().is_gpu()) {
    GTEST_SKIP() << "skip the test because oneMKL GEMM on gpu may give NaN "
                    "(under investigation)";
}
```